src/ExpressionEvaluator/Core/Source/FunctionResolver/FunctionResolverBase.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Roslyn.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection.Metadata;
+
+namespace Microsoft.CodeAnalysis.ExpressionEvaluator
+{
+    internal abstract class FunctionResolverBase<TProcess, TModule, TRequest>
+        where TProcess : class
+        where TModule : class
+        where TRequest : FunctionResolutionRequestBase<TModule>
+    {
+        internal void EnableResolution(TProcess process, TRequest request)
+        {
```

Comment:
We'll want to add a check that errors out of the line offset is non-zero and the language is C# or VB.  Here's the old code for this:
```C#
    if (request.LineOffset != 0)
    {
        // Because of the way filtering works, this method gets called even if we're native debugging.  The native EE supports non-zero line offsets,
        // even though the managed EE doesn't.  Suppress the error for non-VB and C# languages to avoid interfering with native.
        if ((request.CompilerId.LanguageId == DkmLanguageId.CSharp) || (request.CompilerId.LanguageId == DkmLanguageId.VB))
        {
            DkmCompletionRoutine<DkmOnResolverMessageAsyncResult> onComplete = delegate(DkmOnResolverMessageAsyncResult result) { };
            request.OnResolverMessage(workList, DkmBreakpointMessageLevel.StandardError, ShimEEResources.Error_NoLineOffset, onComplete);
            return;
        }
    }
```

The existing error message is
```
Function breakpoints are only supported on the first line
```